src/Content/Version.php

```diff
@@ -319,17 +376,14 @@ protected function prepareFieldsForContent(
 	 */
 	public function publish(Language|string $language = 'default'): void
 	{
-		if ($this->id->is(VersionId::latest()) === true) {
-			throw new LogicException(
-				message: 'This version is already published'
-			);
-		}
-
 		$language = Language::ensure($language);
 
 		// the version needs to exist
 		$this->ensure($language);
```

Comment:
`Question:` Shouldn't these also belong into the rules class?

Reply:
I thought about this as well. I really think they could and maybe even should.

Reply:
I moved it over from the Version and Storage classes and I think it really cleans up the storage classes a lot more.